lib/fhir/query-utils.js

```diff
@@ -116,7 +116,7 @@ module.exports = (mongo) => {
      */
     transformDate: transformDate,
 
-    tokenToSystemValueElemMatch: (fieldToMatch, token) => {
+    tokenToSystemValueElemMatch: (fieldToMatch, token, shouldNotElemMatch) => {
```

Comment:
This is a little confusing given the title of the function. When would we not want to elemMatch over wanting to elemMatch. Maybe we don't ever need to elemMatch as then won't it not return results that have other properties (e.g. text) in the element?

Reply:
I originally had this as two separate function, but a lot of the code was duplicated, so i ended up merging them into one function, but do agree, its a bit confusing.

The problem is that currently the function assumes the token to be searched is an array (whcih is most of the cases - DocumentManifest.identifier), but within the same resource there is a identifer property which isnt an Array, Its a single identifier (DocumentManifest.related.identifier)

elemMatch is needed when matching within an Array, but not when its an object. I had to separate these two to get the expected result returning. Unless there is a way to achieve the same result in both cases?